content/data-loading.md

```diff
@@ -0,0 +1,556 @@
+
+However, it's generally a good idea to use a layout template (which you wrap all your templates in) to subscribe to this subscription anyway. It's better to be consistent about such things, and it makes for a more flexible system if you ever decide you have a screen that *doesn't* need that data.
+
+
+## Patterns for data loading
+
+Across Meteor applications, there are some common patterns of data loading and and management on the client side that are worth knowing. We'll go into more detail about some of these in the {% post_link ui-ux "UI/UX Article" %}.
+
+### Subscription readiness
+
+A key thing to keep in mind is that a subscription will not instantly provide it's data. There'll be a latency between subscribing to the data on the client and it arriving from the publication on the server (and keep in that this time may be a lot longer for your users in production that for you locally in development!)
+
+Although the Tracker system means you often don't *need* to think too much about this in building your apps, usually if you want to get the user experience right, you'll need to know when the data is ready. 
+
+To find that out, `Meteor.subscribe()` and (`this.subscribe()` in templates) returns a "subscription handle", which contains a reactive data source called `.ready()`:
+
+```js
+const handle = Meteor.subscribe('lists/public');
+Tracker.autorun(() => {
+  console.log(`Handle is ${handle.ready() ? 'ready' : 'not ready'}`);  
+});
+```
+
+We can use this information to be more subtle about when we try and show data to users, and when we show a loading screen.
+
+### Changing subscription arguments
+
+We've seen an example already of using an `autorun` to re-subscribe when the (reactive) arguments to a subscription change. It's worth digging in a little more detail to understand what happens in this scenario.
+
+```js
+Template.listsShowPage.onCreated(function() {
+  this.state = new ReactiveDict();
+  this.autorun(() => {
+    this.state.set('listId', FlowRouter.getParam('_id'));
+    this.subscribe('list/todos', this.state.get('listId'));
+  });
+});
+```
+
+In our example, the `autorun` will re-run whenever `this.state.get('listId')` changes, (ultimately because `FlowRouter.getParam('_id')` changes), although other common reactive data sources are:
+
+1. Template data contexts (which you can access reactively with `Template.currentData()`)
+2. The current user status (`Meteor.user()` and `Meteor.loggingIn()`)
+3. The contents of other application specific client data stores.
+
+Technically, what happens when one of these reactive sources changes is the following:
+
+1. The reactive data source *invalidates* the autorun computation (tells it to re-run itself)
```

Comment:
"tells it to re-run itself" -> "Marks it so that it is re-run in the next flush cycle"